.github/workflows/docker_build_publish.yaml

```diff
@@ -21,44 +21,60 @@ on:
     # Publish only on `master`
     branches:
       - master
-
+  pull_request:
```

Comment:
If I understand this correctly, it will trigger the build and push at each pull request creation rather than when a commit is pushed? That would be quite a few docker images with unreviewed code?

Reply:
It'll trigger the build whenever there is a push on master, when there is a new release or when there is a new pull request against master (and anytime that branch is updated after the PR is created)

In the build and push action I prevent pushing to the repository if the event is a pull request, so it should only be used to check if the docker file modifications/github actions modifications are valid without creating a new image on dockerhub.

Also it should only be triggering the CI if the PR includes a file that modifies the docker file or the github action

Reply:
Thanks, just saw the check added in push section. Makes sense.

Reply:
There have been no release triggered in github for many versions
https://github.com/apache/gobblin/releases


Because the docker image build is now tied to a release, there are actually no docker images available 
https://hub.docker.com/r/apache/gobblin/tags

Please create a release for the latest (0.16.0) so we can get a proper container built

https://issues.apache.org/jira/browse/GOBBLIN-1555